usaspending_api/awards/models/transaction_normalized.py

```diff
@@ -0,0 +1,187 @@
+        blank=True, null=True, help_text="The date this transaction was certified"
+    )
+    create_date = models.DateTimeField(
+        auto_now_add=True, blank=True, null=True, help_text="The date this transaction was created in the API"
+    )
+    update_date = models.DateTimeField(
+        auto_now=True,
+        null=True,
+        help_text="The last time this transaction was updated in the API",
+        db_index=True,
+    )
+    fiscal_year = models.IntegerField(
+        blank=True, null=True, help_text="Fiscal Year calculated based on Action Date"
+    )
+    transaction_unique_id = models.TextField(
+        blank=False, null=False, default="NONE", verbose_name="Transaction Unique ID"
+    )
+    generated_unique_award_id = models.TextField(
+        blank=False, null=False, default="NONE", verbose_name="Generated Unique Award ID"
+    )
+    is_fpds = models.BooleanField(blank=False, null=False, default=False, verbose_name="Is FPDS")
+    funding_amount = models.DecimalField(
+        max_digits=23,
+        decimal_places=2,
+        blank=True,
+        null=True,
+        help_text="Assistance data variable.  non_federal_funding_amount + " "federal_action_obligation",
+    )
+    non_federal_funding_amount = models.DecimalField(
+        max_digits=23, decimal_places=2, blank=True, null=True, help_text="Assistance Data variable."
+    )
+    unique_award_key = models.TextField(null=True, db_index=True)  # From broker.
+
+    def __str__(self):
+        return "%s award: %s" % (self.type_description, self.award)
+
+    def newer_than(self, dct):
+        """Compares age of this instance to a Python dictionary
+
+        Determines the age of each by last_modified_date, if set,
+        otherwise action_date.
+        Returns `False` if either side lacks a date completely.
+        """
+
+        my_date = self.last_modified_date
+        their_date = dct.get("last_modified_date")
+        if my_date and their_date:
+            return my_date > their_date
+        else:
+            return False
+
+    @classmethod
+    def get_or_create_transaction(cls, **kwargs):
+        """Gets and updates, or creates, a Transaction
+
+        Transactions must be unique on Award, Awarding Agency, and Mod Number
+        """
+        transaction = (
+            cls.objects.filter(
+                award=kwargs.get("award"), modification_number=kwargs.get("modification_number")
+            )
+            .order_by("-update_date")
+            .first()
+        )
+        if transaction:
+            if not transaction.newer_than(kwargs):
+                for (k, v) in kwargs.items():
+
+                    setattr(transaction, k, v)
+
+            return transaction
+        return cls(**kwargs)
+
+    def save(self, *args, **kwargs):
+        self.fiscal_year = fy(self.action_date)
+        super().save(*args, **kwargs)
+
+    class Meta:
+        db_table = "transaction_normalized"
+        index_together = ["award", "action_date"]
```

Comment:
Why is there a completely new object for this?

Reply:
As discussed, the model file was split up.